modyn/selector/selector.py

```diff
@@ -93,7 +93,12 @@ def get_number_of_samples(self, trigger_id: int) -> int:
         if trigger_id not in self._trigger_size_cache:
             raise ValueError(f"Trigger ID {trigger_id} does not exist!")
 
-        return self._trigger_size_cache[trigger_id]
+        # if we downsample the data, the dataset size is smaller
+        downsampling_scale = (
+            self._strategy.get_downsampling_scale() if isinstance(self._strategy, AbstractDownsampleStrategy) else 1
+        )
```

Comment:
hmm, why is this required? `get_number_of_samples` is currently expected in the code to return the size of the trigger training set. We then later have a downsampled trigger training set which is indeed smaller (and we need to log this and remove it from the trainer server at some point), but I am not sure whether this should be done here.

Reply:
The idea is that the status bar is based on the value of get_number_of_samples which returns the number of samples **we are training on**. If we use downsampling, the number of datapoints actually used for training is lower and we have to take this into account (otherwise, for example, the status bar never completes).

Reply:
still open (added another comment)